runtime/gc_base/modronapi.cpp

```diff
@@ -1150,6 +1150,16 @@ j9gc_reinitialize_for_restore(J9VMThread *vmThread, const char **nlsMsgFormat)
 		goto _error;
 	}
 
+#if defined(OMR_GC_MODRON_SCAVENGER)
+	if (extensions->scavengerEnabled && (NULL != extensions->scavenger)) {
+		if (!extensions->scavenger->reinitializeForRestore(env)) {
+			*nlsMsgFormat = j9nls_lookup_message(J9NLS_DO_NOT_PRINT_MESSAGE_TAG | J9NLS_DO_NOT_APPEND_NEWLINE,
+					J9NLS_GC_FAILED_TO_INITIALIZE_OUT_OF_MEMORY, NULL);
```

Comment:
OOM error!?

Reply:
Its a generic error message. It is technically correct as Scavenger reinit will only fail if alloc of sublists fails, which will be due to OOM.

```
J9NLS_GC_FAILED_TO_INITIALIZE_OUT_OF_MEMORY=Failed to initialize; out of memory
# START NON-TRANSLATABLE
J9NLS_GC_FAILED_TO_INITIALIZE_OUT_OF_MEMORY.explanation=The garbage collector failed to initialize due to an out of memory condition
J9NLS_GC_FAILED_TO_INITIALIZE_OUT_OF_MEMORY.system_action=The JVM will terminate
J9NLS_GC_FAILED_TO_INITIALIZE_OUT_OF_MEMORY.user_response=Ensure the system has sufficient memory available
```

Reply:
But I suppose this is not relevant with your suggestion of moving scavenger reinit inside of ParallelGlobalGC::reinitializeForRestore